README.md

```diff
@@ -23,396 +23,373 @@ Stable release versions are manually released from the main branch.
-    unbondingScript,
-    slashingScript,
-    dataEmbedScript
-  },
-  stakingAmount,
-  changeAddress,
-  inputUTXOs,
-  network(),
-  feeRate,
-  btcWallet.isTaproot ? btcWallet.publicKeyNoCoord() : undefined,
-  lockHeight,
-);
-
-const signedStakingPsbt = await btcWallet.signPsbt(unsignedStakingPsbt.psbt.toHex());
-const stakingTx = Psbt.fromHex(signedStakingPsbt).extractTransaction();
+export interface BtcProvider {
+  // Sign a PSBT
+  signPsbt(signingStep: SigningStep, psbtHex: string): Promise<string>;
+  // Sign a message using the ECDSA type
+  // This is optional and only required if you would like to use the 
+  // `createProofOfPossession` function
+  signMessage?: (
+    signingStep: SigningStep, message: string, type: "ecdsa" || "bip322-simple"
+  ) => Promise<string>;
+}
+
+export interface BabylonProvider {
+  signTransaction: <T extends object>(
+    signingStep: SigningStep,
+    msg: {
+      typeUrl: string;
+      value: T;
+    }
+  ) => Promise<Uint8Array>
+}
 ```
 
-Public key is needed only if the wallet is in Taproot mode, for `tapInternalKey`.
+### Post-Staking Registration
 
-### Create unbonding transaction
+This refers to registration of an already existing BTC staking transaction into 
+Babylon chain network. 
+For more details, refer to the [Babylon node documentation](https://github.com/babylonlabs-io/babylon/blob/release/v1.x/docs/register-bitcoin-stake.md#21-post-staking-registration).
 
-The staking script allows users to on-demand unbond their locked stake before
-the staking transaction timelock expires, subject to an unbonding period.
-
-The unbonding transaction can be created as follows:
+> **Note**: This method focuses on creating Babylon chain transactions, 
+> not Bitcoin transactions. Specifically:
+>
+> It creates Babylon transactions that contain:
+>   - The existing on-chain Bitcoin staking transaction
+>   - Unsigned unbonding Bitcoin transaction
+>   - Signed Slashing Bitcoin transactions
+>   - Proof of possession
 
 ```ts
-import { unbondingTransaction } from "@babylonlabs-io/btc-staking-ts";
-import { Psbt, Transaction } from "bitcoinjs-lib";
-
-// Unbonding fee in satoshis. number
-const unbondingFee: number = 500;
-
-const unsignedUnbondingPsbt: {psbt: Psbt} = unbondingTransaction(
-  scripts: {
-    unbondingScript,
-    unbondingTimelockScript,
-    timelockScript,
-    slashingScript,
-  },
-  stakingTx,
-  unbondingFee,
-  network,
+import {
+  BabylonBtcStakingManager,
+} from "@babylonlabs-io/btc-staking-ts";
+
+const manager = new BabylonBtcStakingManager(
+  btcNetwork,
+  stakingParams,
+  btcProvider,
+  bbnProvider,
 );
 
-const signedUnbondingPsbt = await signPsbt(unsignedUnbondingPsbt.psbt.toHex());
-const unbondingTx = Psbt.fromHex(signedUnbondingPsbt).extractTransaction();
+const { 
+  signedBabylonTx
+} = await manager.postStakeRegistrationBabylonTransaction(
+  stakerInfo,
+  stakingTx,
+  stakingHeight,
+  stakingInput,
+  inclusionProof,
+  bech32Address,
+);
```

Comment:
This code listing does not have any comments. What does it do and how does it refer to the previous descriptions you have provided?